client/task_runner.go

```diff
@@ -433,9 +457,60 @@ func (r *TaskRunner) startTask() error {
 	r.handleLock.Lock()
 	r.handle = handle
 	r.handleLock.Unlock()
+	r.stopResourceMonitorCh = make(chan struct{})
+	go r.monitorUsage()
 	return nil
 }
 
+// monitorUsage starts collecting resource usage stats of a Task
+func (r *TaskRunner) monitorUsage() {
+	for {
+		next := time.NewTimer(1 * time.Second)
+		select {
+		case <-next.C:
+			ru, err := r.handle.Stats()
+			if err != nil {
+				r.logger.Printf("[DEBUG] client.taskrunner: error fetching stats of task %v: %v", r.task.Name, err)
+			}
+			r.resourceUsageLock.RLock()
+			r.resourceUsage.Enqueue(ru)
+			r.resourceUsageLock.RUnlock()
+			next.Reset(1 * time.Second)
+		case <-r.stopResourceMonitorCh:
+			next.Stop()
+			return
+		}
+	}
+}
+
+// TaskStatsReporter returns the stats reporter of the task
+func (r *TaskRunner) StatsReporter() TaskStatsReporter {
+	return r
+}
+
+// ResourceUsage returns the last resource utilization datapoint collected
+func (r *TaskRunner) ResourceUsage() *cstructs.TaskResourceUsage {
+	r.resourceUsageLock.RLock()
+	defer r.resourceUsageLock.RUnlock()
+	val := r.resourceUsage.Peek()
```

Comment:
If `Peek()` returns `nil`, then typecasting it should also return nil and the API would return nil.